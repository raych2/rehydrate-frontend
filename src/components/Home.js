import { Link } from 'react-router-dom';
import logo from '../assets/Rehydrate_Final_white_small2.png';

const Home = () => {
  return (
    <div className="home">
      <h1 className="title">Rehydrate</h1>
      <img className="logo" alt="rehydrate logo" src={logo}></img>
      <Link to="about" className="learn">
        Learn More!
      </Link>
    </div>
  );
};

export default Home;
