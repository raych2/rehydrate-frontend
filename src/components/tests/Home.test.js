import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Router, MemoryRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Home from '../Home';

describe('Home component', () => {
  test('renders correctly', () => {
    const { container } = render(<Home />, { wrapper: MemoryRouter });
    expect(container).toMatchSnapshot();
  });
  test('navigates to About page after link is clicked', async () => {
    const history = createMemoryHistory();
    history.push = jest.fn();
    render(
      <Router location={history.location} navigator={history}>
        <Home />
      </Router>
    );
    const user = userEvent.setup();
    await user.click(screen.getByRole('link', { name: 'Learn More!' }));
    expect(history.push).toHaveBeenCalledWith(
      {
        hash: '',
        pathname: '/about',
        search: '',
      },
      undefined
    );
  });
});
