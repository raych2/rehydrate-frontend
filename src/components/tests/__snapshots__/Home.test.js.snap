// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div>
  <div
    class="home"
  >
    <h1
      class="title"
    >
      Rehydrate
    </h1>
    <img
      alt="rehydrate logo"
      class="logo"
      src="Rehydrate_Final_white_small2.png"
    />
    <a
      class="learn"
      href="/about"
    >
      Learn More!
    </a>
  </div>
</div>
`;
